scala.html.markdown

```diff
@@ -406,8 +407,63 @@ val otherGeorge = george.copy(phoneNumber = "9876")
 // And many others. Case classes also get pattern matching for free, see below.
 
 
-// Traits coming soon!
+// Traits
+Traits are somewhat similar to Java interfaces. With traits you define  
+a type and method signatures. Other than Java interfaces, Scala allows  
+you to mix method signatures with method implementations.  
+Parameters for constructors are not allowed.  
+Traits can inherit from other traits or classes, but only if they have  
+no parameters.  
+```
+trait Bird {
+	def breed: String
+	def color: String
+	def saySomething: String = "tweet"
+	def canFly: Boolean
+}
+class Duck extends Bird{
+	val breed = "bird"
+	val color = "green"
+	def canFly = true
+}
+```
+  
+```
+scala> var d = new Duck
```

Comment:
Use `val` whenever appropriate. Also, rest of this tutorial doesn't show output in the REPL style. We just do
    
    val d = new Duck
    d.breed  // "bird"